webapp/platform/client/src/websocket.ts

```diff
@@ -407,6 +444,21 @@ export default class WebSocketClient {
         }
     }
 
+    ping(responseCallback?: (msg: any) => void) {
+        const msg = {
+            action: 'ping',
+            seq: this.responseSequence++,
+        };
+
+        if (responseCallback) {
+            this.responseCallbacks[msg.seq] = responseCallback;
```

Comment:
Just curiosity, but how are we executing the responseCallback? I don't remember that part of the code.

Reply:
This was already implemented on the webapp websocket client, but it doesn't exist in the mobile clients. `responseCallbacks` here is a map from message sequence number to a callback function. When we get a response, we'll look at the map for a function associated with the sequence number of that response, and execute it. Then that function is dropped from the map.

Reply:
There has been a history about sharing code between mobile and web.

Long story short... it is complicated because mobile devs and web devs have been more or less siloed, so changes are made to the common library thinking only about web, and that crashes mobile (or crashes mobile's backwards compatibility).

From time to time there are conversations about this (I would say @rahimrahman is one of the promoters right now), but I am quite skeptic about being able to properly share code between the two systems.